### DISCUSSION TITLE
SQL Statement: support export and import table data from file in filesystem

### DISCUSSION THREAD

**Discussion by aceforeverd** [4paradigm/OpenMLDB#494] [General]
# Brief

- write query into a file, e.g a csv format file #455 
- read and insert table data from file #443 

*only table data is imported, not schema*

# Well-known mechanism
MySQL's [select into outfile](https://mariadb.com/kb/en/select-into-outfile/) & [load data infile](https://dev.mysql.com/doc/refman/8.0/en/load-data.html)

# Proposal (Draft)

```
<select into outfile> ::= <query> INTO OUTFILE 'file_name' OPTIONS <option_list>

<load data infile> ::= LOAD DATA INFILE 'file_name' INTO TABLE 'table_name' OPTIONS <options_list>

<options_list> ::= ( key = value , ... )
```
 
## Predefined options
- data file related:
  - format, e.g `CSV`
  - delimit
  - charset
- load process related:
  - replace strategy: `REPLACE` or `IGNORE`
  - timeout

*since options is not well defined yet, they were configured in `<options_list>` for flexibility*

**Comment by jingchen2222** [comment]
> ## Predefined options
> * data file related:
>   
>   * format, e.g `CSV`
>   * delimit
>   * charset
> ...
> _since options is not well defined yet, they were configured in `<options_list>` for flexibility_

predefine LINES delimit and FIELDS delimit ?

**Reply by aceforeverd** [reply]
MySQL has
```
    [{FIELDS | COLUMNS}
        [TERMINATED BY 'string']
        [[OPTIONALLY] ENCLOSED BY 'char']
        [ESCAPED BY 'char']
    ]
    [LINES
        [STARTING BY 'string']
        [TERMINATED BY 'string']
    ]
```

if defined in options list, can write as `fileds_terminated = '', lines_terminated = ''`
I'm not sure if those options should put into SQL directly
